apps/els_lsp/src/erlang_ls.erl

```diff
@@ -96,7 +108,24 @@ set(log_dir, Dir) ->
 set(log_level, Level) ->
   application:set_env(els_core, log_level, list_to_atom(Level));
 set(port_old, Port) ->
-  application:set_env(els_core, port, Port).
+  application:set_env(els_core, port, Port);
+set(wrangler_enabled, Opt) ->
+  application:set_env(els_core, wrangler_enabled, Opt);
+set(wrangler_dir, "undefined") -> ok;
+set(wrangler_dir, Dir) ->
+  %% FIXME can't log since logging is not configured yet
```

Comment:
configure_logging() is called after parse_args(). What do you suggest?

Reply:
In https://github.com/erlang-ls/erlang_ls/pull/1148/files there's an attempt at extending the `els_diagnostics` behaviour with an `init` function. We could use that approach and have a callback module for Wrangler. In the `init`you could then initialize Wrangler.

Reply:
Since we may want to extend not only code actions but code lenses (and semantic highlights), we need to initialize wrangler in general (and once only, when the extension is loaded). That is different from what you have suggested, right?